documentation/modules/exploit/unix/webapp/drupal_drupalgeddon2.md

```diff
@@ -0,0 +1,101 @@
+## Intro
+
+> This module exploits a Drupal property injection in the Forms API.
+> Drupal 6.x, < 7.58, 8.2.x, < 8.3.9, < 8.4.6, and < 8.5.1 are
+> vulnerable.
+
+## Setup
+
+Use the provided Docker images here: <https://hub.docker.com/_/drupal/>.
+
+Tested on the 7.57 and 8.4.5 tags (versions).
+
+## Targets
+
+```
+Id  Name
+--  ----
+0   Automatic (PHP In-Memory)
+1   Automatic (PHP Dropper)
+2   Automatic (Unix In-Memory)
+3   Automatic (Linux Dropper)
+4   Drupal 7.x (PHP In-Memory)
+5   Drupal 7.x (PHP Dropper)
+6   Drupal 7.x (Unix In-Memory)
+7   Drupal 7.x (Linux Dropper)
+8   Drupal 8.x (PHP In-Memory)
+9   Drupal 8.x (PHP Dropper)
+10  Drupal 8.x (Unix In-Memory)
+11  Drupal 8.x (Linux Dropper)
+```
+
+Automatic targeting means the Drupal version will be detected first.
+Targets with a specific version will do as they're told (regardless of
+what the server is running).
+
+Dropper targets write to disk. In-memory targets don't. Be mindful of
+showing up in someone's process list, though. A dropper might be more
+viable in that regard.
+
+## Options
+
+**TARGETURI**
+
+Set this to the remote path of the vulnerable Drupal install. Defaults
+to `/` for the web root.
+
+**PHP_FUNC**
+
+Set this to the PHP function you'd like to execute. Defaults to
+`passthru`.
+
+**DUMP_OUTPUT**
+
+Enable this if you'd like to see HTTP responses, including command
+output. Defaults to `false` unless `cmd/unix/generic` is your payload.
+
+**VERBOSE**
+
+Enable this to show what function and command were executed. Defaults to
+`false` due to the sometimes excessive output.
+
+**ForceExploit**
+
+Enable this to force exploitation regardless of the check result.
+Defaults to `false`, meaning the check result is respected.
+
+**WritableDir**
+
+Set this to a writable directory without `noexec` for binary payloads.
+Defaults to the current working directory (usually the webapp root).
+
+## Usage
+
+```
```

Comment:
can you add a h3 here with the exact drupal version thats being tested?  Most likely `### Drupal 7.57 on docker image`

Reply:
Sure thing!

Reply:
I dropped the header, since it stuck out too proudly. I've written in the tested version instead. Thoughts?